src/main/java/com/bitvavo/api/Bitvavo.java

```diff
@@ -714,43 +605,34 @@ public JSONArray depositHistory(JSONObject options) {
    * @return JSONArray response, get individual withdrawals by iterating over the array: response.getJSONObject(index)
    */
   public JSONArray withdrawalHistory(JSONObject options) {
-    String postfix = createPostfix(options);
-    return privateRequestArray("/withdrawalHistory", postfix, "GET", new JSONObject());
+    String urlParams = createUrlParams(options);
+    return privateRequestArray("/withdrawalHistory", urlParams, "GET", new JSONObject());
   }
 
   /**
    * Creates a websocket object
    * @return Websocket the object on which all websocket function can be called.
    */
   public Websocket newWebsocket() {
-    websocketObject = new Websocket();
-    return websocketObject;
+    return new Websocket();
   }
 
   public class Websocket {
     public Websocket() {
       try {
         final WebsocketClientEndpoint clientEndPoint = new WebsocketClientEndpoint(new URI(Bitvavo.this.wsUrl), Bitvavo.this);
-        clientEndPoint.addAuthenticateHandler(new WebsocketClientEndpoint.MessageHandler() {
-          public void handleMessage(JSONObject response) {
-            if(response.has("authenticated")) {
-              authenticated = true;
-              debugToConsole("We registered authenticated as true");
-            }
-          }
-        });
-        clientEndPoint.addMessageHandler(new WebsocketClientEndpoint.MessageHandler() {
-          public void handleMessage(JSONObject response) {
-            errorToConsole("Unexpected message: " + response);
+        clientEndPoint.addAuthenticateHandler(response -> {
+          if (response.has("authenticated")) {
+            authenticated = true;
+            debugToConsole("We registered authenticated as true");
           }
         });
+        clientEndPoint.addMessageHandler(response -> errorToConsole("Unexpected message: " + response));
         ws = clientEndPoint;
-        book = new HashMap<String,Object>();
-        KeepAliveThread keepAliveThread = new KeepAliveThread();
+        book = new HashMap<>();
+        Bitvavo.this.keepAliveThread = new KeepAliveThread();
```

Comment:
Is this thing ever started?

When creating new threads they may be daemon threads by default, completely negating the need for this